Mage.Sets/src/mage/cards/g/GleefulDemolition.java

```diff
@@ -0,0 +1,81 @@
+package mage.cards.g;
+
+import java.util.UUID;
+
+import mage.abilities.Ability;
+import mage.abilities.effects.Effect;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.DestroyTargetEffect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.filter.StaticFilters;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.game.permanent.token.PhyrexianGoblinToken;
+import mage.game.permanent.token.Token;
+import mage.players.Player;
+import mage.target.TargetPermanent;
+import mage.target.common.TargetArtifactPermanent;
+
+/**
+ *
+ * @author AhmadYProjects
+ */
+public final class GleefulDemolition extends CardImpl {
+
+    public GleefulDemolition(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.SORCERY}, "{R}");
+        
+
+        // Destroy target artifact. If you controlled that artifact, creature three 1/1 red Phyrexian Goblin creature tokens.
+        this.getSpellAbility().addTarget(new TargetPermanent(StaticFilters.FILTER_PERMANENT_ARTIFACT));
+        this.getSpellAbility().addEffect(new GleefulDemolitionEffect());
+        this.getSpellAbility().addEffect(new DestroyTargetEffect());
+
+    }
+
+    private GleefulDemolition(final GleefulDemolition card) {
+        super(card);
+    }
+
+    @Override
+    public GleefulDemolition copy() {
+        return new GleefulDemolition(this);
+    }
+}
+
+class GleefulDemolitionEffect extends DestroyTargetEffect {
```

Comment:
this doesn't really need to extend `DestroyTargetEffect`, it can just be a `OneShotEffect`